### PR TITLE
feat(iFrame): Emit event when recording status changes, including errors

### PR DESCRIPTION
**Pull request by izakgl**
A little while ago I created a similar pull request but after a bit of thought I realized I could improve it further. This pull request features an event emitted to external api when the recording status changes, but it also includes the errors (which previous pull request didn't).

If there are any improvements to be made, please do tell.

### CODE REVIEW

**Comment by saghul** [comment]
This needs a rebase.

**Comment by izakgl** [comment]
Rebase done

**Comment by damencho** [comment]
Jenkins, test this please.

**Comment by damencho** [comment]
@izakgl Can you add documentation for it in https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-iframe, the repo is here: https://github.com/jitsi/handbook/blob/master/docs/dev-guide/iframe.md

File: react/features/recording/middleware.js [hunk 1]
```diff
@@ -181,6 +182,8 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
                 if (soundID) {
                     dispatch(playSound(soundID));
                 }
+
+                APP.API.notifyRecordingStatusChanged(true, mode);
```
Comment (damencho) [inline react/features/recording/middleware.js:186]: @saghul is this ok for mobile?

Comment (saghul) [reply react/features/recording/middleware.js:186]: No, mobile doesn't have the APP global.

**Review by damencho — CHANGES_REQUESTED** [review]
The changes will break mobile, as Saul mentioned, APP is not available there ...

File: react/features/recording/middleware.js (hunk 1 above)
Comment (izakgl) [reply react/features/recording/middleware.js:186]: I should be checking that typeof APP is not undefined, correct?
```javascript
if (typeof APP !== 'undefined')
```

**Review by damencho — APPROVED** [review]

**Comment by damencho** [comment]
Jenkins, test this please.

**Pull request merged**